web_client/models/WidgetModel.js

```diff
@@ -32,6 +25,13 @@ var WidgetModel = Backbone.Model.extend({
     },
 
     /**
+     * Sets initial model attributes with normalization.
+     */
+    initialize: function (model) {
+        this.set(_.defaults(model || {}, this.defaults));
```

Comment:
This entire `initialize` function is unnecessary and can be removed. All of this work is already [done by Backbone](http://backbonejs.org/docs/backbone.html#section-54) when a new Model is instantiated.